pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1130,8 +1130,8 @@ def _looks_like_infinite_iterator(param: nodes.NodeNG) -> bool:
                 return inferred.qname() in KNOWN_INFINITE_ITERATORS
             return False
 
-        if isinstance(node.func, nodes.Attribute):
-            # A next() method, which is now what we want.
+        if isinstance(node.func, nodes.Attribute) or not node.args:
```

Comment:
Why don't we fix L1142? That if seem to disregard the `len(args) == 0` case?

Reply:
The way the code is written is pretty subtle.. let me reproduce and explain:
```
            has_sentinel_value = len(node.args) > 1
            if (
                isinstance(frame, nodes.FunctionDef)
                and frame.is_generator()
                and not has_sentinel_value
                and not utils.node_ignores_exception(node, StopIteration)
                and not _looks_like_infinite_iterator(node.args[0])
            ):
```
`has_sentinel_value` would be true if there are 2+ args, and False if there are 0 or 1 args. In our specific case of args 0,  because this check `and not has_sentinel_value` is True, then we continue on to check  the other two checks of the if statement and hit `and not _looks_like_infinite_iterator(node.args[0])`, which will fail because we just said args is len 0.

Adding ` or not node.args:` higher up just short circuits everything cleanly.

If you have nicer code that would perform the same thing, do let me know.

Reply:
Okay, but this changes still doesn't really fix the issue that is described. It just prevents the issue from happening by checking for another heuristic.

Why don't we actually check the `qname` of `inferred` to be `builtins.next`? That seems much more robust?

Reply:
Yeah I think an `isinstance` check and then a `qname` call on that line should be doable!

Reply:
I thought it was impossible to do that because of this: https://github.com/PyCQA/pylint/pull/7622#discussion_r997001339